### PR TITLE
fix(query): Fix compressed buf not consumed correctly

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

fix(query): Fix compressed buf not consumed correctly

## Changelog

- Bug Fix

### CODE REVIEW

File: query/src/servers/http/v1/multipart_format.rs [hunk 1]
```diff
@@ -282,13 +282,15 @@ impl Processor for SequentialInputFormatSource {
                 let data = match &mut self.input_decompress {
                     None => data,
                     Some(decompress) => {
```
Comment (sundy-li) [inline query/src/servers/http/v1/multipart_format.rs:284]: Is there a `decompress_bound` method for each compressor?

Comment (Xuanwo) [reply query/src/servers/http/v1/multipart_format.rs:284]: What's `decompress_bound` for?

Comment (sundy-li) [reply query/src/servers/http/v1/multipart_format.rs:284]: https://docs.rs/zstd-safe/4.1.1+zstd.1.5.0/zstd_safe/fn.decompress_bound.html

Comment (Xuanwo) [reply query/src/servers/http/v1/multipart_format.rs:284]: > Is there a `decompress_bound` method for each compressor?

As far as I know, `async-compression` doesn't expose this method. Maybe we can hack it if needed.

**Review by zhang2014 — APPROVED** [review]

**Comment by BohuTANG** [comment]
```
<Backtrace disabled by default. Please use RUST_BACKTRACE=1 to enable> 
[ FAIL ] - result differs with:
--- /home/runner/work/databend/databend/tests/suites/1_stateful/00_copy/00_0001_upload_to_stage.result	2022-06-02 09:22:34.945052455 +0000
+++ /home/runner/work/databend/databend/tests/suites/1_stateful/00_copy/00_0001_upload_to_stage.stdout	2022-06-02 09:25:55.253876497 +0000
@@ -1 +1,2 @@
-00_0001_upload_to_stage.sh
+ERROR 1105 (HY000) at line 1: Code: 1068, displayText = Cannot join handle from context's runtime, cause: task 3180 panicked.
+ERROR 1105 (HY000) at line 1: Code: 1068, displayText = Cannot join handle from context's runtime, cause: task 3205 panicked.
```

**Comment by Xuanwo** [comment]
> ```
> <Backtrace disabled by default. Please use RUST_BACKTRACE=1 to enable> 
> [ FAIL ] - result differs with:
> --- /home/runner/work/databend/databend/tests/suites/1_stateful/00_copy/00_0001_upload_to_stage.result	2022-06-02 09:22:34.945052455 +0000
> +++ /home/runner/work/databend/databend/tests/suites/1_stateful/00_copy/00_0001_upload_to_stage.stdout	2022-06-02 09:25:55.253876497 +0000
> @@ -1 +1,2 @@
> -00_0001_upload_to_stage.sh
> +ERROR 1105 (HY000) at line 1: Code: 1068, displayText = Cannot join handle from context's runtime, cause: task 3180 panicked.
> +ERROR 1105 (HY000) at line 1: Code: 1068, displayText = Cannot join handle from context's runtime, cause: task 3205 panicked.
> ```

This bug related to #4491, I will fix them at the same time

**Comment by Xuanwo** [comment]
https://github.com/datafuselabs/opendal/pull/327 will fix this issue.

**Comment by wubx** [comment]
Should we merge it?

**Comment by BohuTANG** [comment]
> test_stateful_standalone

It's not ready yet, there are some stateful-test failed and related to this PR:
```
The failure tests:
    /home/runner/work/databend/databend/tests/suites/1_stateful/00_copy/00_0000_copy_from_s3_location.sh
    /home/runner/work/databend/databend/tests/suites/1_stateful/00_copy/00_0000_copy_from_stage.sh
```

File: query/src/servers/http/v1/stage.rs [hunk 2]
```diff
@@ -49,6 +48,8 @@ pub async fn upload_to_stage(
 
     let user_mgr = context.get_user_manager();
 
+    // TODO(xuanwo): logic here seems buggy, we need to fix it.
+    // It's incorrect to get operator from context if we are uploading an external stage.
```
Comment (BohuTANG) [inline query/src/servers/http/v1/stage.rs:52]: Add new function as `get_stage_operator`?

**Comment by Xuanwo** [comment]
I will refactor path-related logic after this bug fix.

File: query/src/servers/http/v1/stage.rs (hunk 2 above)
Comment (Xuanwo) [reply query/src/servers/http/v1/stage.rs:51]: These logics are not covered by stateful tests. So I think they are wrong for a long time.

I will fix them in another along with stateful tests.

**Comment by Xuanwo** [comment]
@BohuTANG This PR is ready to go

**Comment by wubx** [comment]
Awesome.
```
ls ./dataset/*.csv.gz|xargs -P 8 -I{} curl -H "insert_sql:insert into ontime format CSV" -H "skip_header:1"   -H "compression:gzip" -F "upload=@{}" -XPUT http://root:@localhost:8000/v1/streaming_load
```
can work too.

**Pull request merged**